task/backend/scheduler/scheduler.go

```diff
@@ -0,0 +1,78 @@
+package scheduler
+
+import (
+	"context"
+	"time"
+
+	"github.com/influxdata/influxdb"
+)
+
+type Storage interface {
+	// UpdateLatestCompleted will allow us to set a latest completed time for an id.
+	UpdateLatestCompleted(ctx context.Context, id influxdb.ID, lastRun time.Time) error
+
+	// retrieve the latest completed for this schedulable id.
+	// If GetLatestCompleted is empty scheduler will use "now" as its latest completed
+	GetLatestCompleted(ctx context.Context, id influxdb.ID) (time.Time, error)
+}
+
+type Executor interface {
+	// Execute is used to execute run's for any schedulable object.
+	// the executor can go through manual runs, clean currently running, and then create a new run based on `now`.
+	// if Now is zero we can just do the first 2 steps (This is how we would trigger manual runs).
+	// Errors returned from the execute request imply that this attempt has failed and
+	// should be put back in scheduler and re executed at a alter time. We will add scheduler specific errors
+	// so the time can be configurable.
+	Execute(ctx context.Context, id influxdb.ID, meta interface{}, now time.Time) (Execution, error)
+
+	CancelRun(ctx context.Context, taskID, runID influxdb.ID) error
+}
+
+type Execution interface {
+	// ID is a unique ID usable to look up or cancel a run
+	ID() influxdb.ID
+
+	// Status can be used to view what the current status of this execution is
```

Comment:
Could the Status() return type be a type other than string? Is the Status() string a user-facing message?  If so, would you mention that it is user-facing in the comments?

Reply:
We have a system now that is a RunStatus int that has a `String` and some json marshaling function. I suspect that would be better here. I suspect the status would be used for logging/metrics. Inside the influxdb system this status would be used in the run list http action.

Reply:
Whatever status ends up being I would like it to be some public type. I think it would make anything implementing the executor interface much cleaner if it didn't have to import the scheduler.